task7/script.js

```diff
@@ -0,0 +1,57 @@
+const documentFragment = document.createDocumentFragment();
+
+const ul = document.getElementById("ul");
+
+const listItem = [{
+  to: "bookmark.html",
+  img: "1.png",
+  alt: "画像1",
+  text: "ブックマーク"
+}, {
+  to: "message.html",
+  img: "2.png",
+  alt: "画像2",
+  text: "メッセージ"
+}];
+
+const getListItem = () => {
+  return new Promise((resolve) => {
+    resolve(listItem);
+    insertLoading();
+  });
+};
+
```

Comment:
I thought we should use 'setTimeout' here. Because the meaning of Task 7 is to show the loading image until it is resolved after 3 seconds I think.

Reply:
I thought ”insertLoading()” would be better outside of ”Promise”. The reason is that the value to be resolved is only for listItem. My idea is below↓

const getListItem = () => {
    insertLoading();
    return new Promise((resolve,reject) => {
    setTimeout(() => {
        resolve(listItem);
    }, 3000);
  });
};

Reply:
僕の考えは、
insertLoading()は、
const getListItem = () => {}のスコープの外がいいと思います。

なぜなら、getListItem()は、値を取得するだけの関数だからです。

なので、insetLoading関数は、外に出すべきです。
その際に、hoistingに注意してください。
意味がわからなければ日本語でも電話でもしてください。

Reply:
My idea is different from Mr. Sakyo.

insertLoading ()
I think it's better outside the scope of const getListItem = () => {}.

Because getListItem () is just a function to get the value.

So the insetLoading function should go out.
At that time, pay attention to hoisting.
If you don't understand the meaning, please call me in Japanese or by phone.

Reply:
Thank you for the review,Ms Sakyo and Mr Yasu.
I moved it in getListitem in 4ec98c9,I thought it was okay to put it out of scope, so I fixed it  in e6715b9.